src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,186 @@
+import TimeField from 'src/components/fields/TimeField';
+import { EventFormWrapper, Form } from './styled';
+import { EVENT_OPTIONS } from './constants';
+
+export type EventDataInput = {
+  id?: string;
+  name?: string;
+  description?: string;
+  startTime?: Date;
+  endTime?: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType?: string;
+};
+
+export type EventDataOutput = {
+  id?: string;
+  name: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  imageUrl?: string;
+  imageAlt?: string;
+  location?: string;
+  eventType: string;
+};
+
+type EventFormProps = {
+  headerTitle: string;
+  onSubmit: (data: EventDataOutput) => Promise<void>;
+} & EventDataInput;
+
+function EventForm({ onSubmit, ...rest }: EventFormProps) {
+  const [name, setName] = useState(rest.name || '');
+  const [imageUrl, setImageUrl] = useState(rest.imageUrl || '');
+  const [imageAlt, setImageAlt] = useState(rest.imageAlt || '');
+  const [date, setDate] = useState(
+    format(
+      new Date((rest.startTime || new Date(Date.now())).toDateString()) ||
+        Date.now(),
+      'yyyy-MM-dd'
+    )
+  );
+  const [startTime, setStartTime] = useState(
+    format(rest.startTime || Date.now(), 'HH:mm')
+  );
+  const [endTime, setEndTime] = useState(
+    format(rest.endTime || Date.now(), 'HH:mm')
+  );
+  const [description, setDescription] = useState(rest.description || '');
+  const [eventType, setEventType] = useState(rest.eventType || '');
+  const [location, setLocation] = useState(rest.location || '');
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [hasSubmitted, setHasSubmitted] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const isFormValid = name && description && eventType;
+  const isSubmitDisabled = !isFormValid || isSubmitting;
+
+  const handleSubmit = useCallback(async () => {
+    setIsSubmitting(true);
+    setError(null);
+
+    try {
+      await onSubmit({
+        id: rest.id,
+        name,
+        description,
+        startTime: new Date(date + ' ' + startTime),
+        endTime: new Date(date + ' ' + endTime),
+        imageUrl,
+        imageAlt,
+        location,
+        eventType,
+      });
+    } catch (error) {
+      setError(error.message);
+    } finally {
+      setIsSubmitting(false);
+    }
+    setHasSubmitted(true);
+  }, [
+    name,
+    description,
+    date,
+    startTime,
+    endTime,
+    imageUrl,
+    imageAlt,
+    location,
+    eventType,
+    setIsSubmitting,
+    setError,
+    onSubmit,
+    rest,
+  ]);
+
+  return (
+    <EventFormWrapper>
+      <h1>{rest.headerTitle}</h1>
+      <Form>
+        <TextField
+          name="name"
+          label="Event name"
+          value={name}
+          onChange={setName}
+        />
+        <DateField
+          name="date"
+          label="Date"
+          value={date}
+          onChange={(date: string) => setDate(date)}
```

Comment:
Can't this just be `onChange={setDate}`?

On a side note, I see that the type of `onChange` has been defined to be `Function`, which is not ideal 😅. We should probably change these the type of all Field component `onChange` prop to some more sensible type, e.g. `(date: string) => void` or `(date: Date) => void` for the `DateField` component.